### PR TITLE
Allow local users to attach Volumes to their wercker containers

### PR DESCRIPTION
**Pull request by faiq**
experiment to see if docker runs in docker

### CODE REVIEW

**Comment by termie** [comment]
good PoC

**Comment by sgoings** [comment]
Some thoughts from the peeps @ Docker: https://github.com/jpetazzo/dind

File: docker/box.go [hunk 1]
```diff
@@ -168,6 +170,9 @@ func (b *DockerBox) binds() ([]string, error) {
 			// volumes[b.options.MntPath(entry.Name())] = struct{}{}
 		}
 	}
+	for _, volume := range b.volumes {
```
Comment (termie) [inline docker/box.go]: keep in mind this has to be behind a check for options.EnableDevSteps, it should only be enable for dev mode

Comment (termie) [inline docker/box.go]: it can be it's own flag, enable-volumes if we want

File: docker/shellstep.go [hunk 2]
```diff
@@ -130,5 +131,5 @@ func (s *ShellStep) ReportPath(...string) string {
 
 // ShouldSyncEnv before running this step = TRUE
 func (s *ShellStep) ShouldSyncEnv() bool {
-	return true
+	return false
```
Comment (termie) [inline docker/shellstep.go]: this can be true again, you'll want to rebase on master

File: docker/shellstep.go [hunk 3]
```diff
@@ -59,10 +59,11 @@ func NewShellStep(stepConfig *core.StepConfig, options *core.PipelineOptions, do
 	})
 
 	return &ShellStep{
-		BaseStep: baseStep,
-		options:  options,
-		data:     stepConfig.Data,
-		logger:   util.RootLogger().WithField("Logger", "ShellStep"),
+		BaseStep:      baseStep,
+		options:       options,
+		dockerOptions: dockerOptions,
```
Comment (termie) [inline docker/shellstep.go]: once you rebase on master this should disappear

File: test-all.sh [hunk 4]
```diff
@@ -51,7 +51,7 @@ testDirectMount() {
 
 runTests() {
   basicTest "source-path" build $testsDir/source-path || return 1
-  basicTest "test local services" build $testsDir/local-service/service-consumer || return 1
+  basicTest "test local services" --debug build  $testsDir/local-service/service-consumer || return 1
```
Comment (termie) [inline test-all.sh]: we might consider just having debug on all the time for these

File: wercker.yml [hunk 5]
```diff
@@ -1,4 +1,6 @@
-box: tcnksm/gox:1.5.1
+box:
+  id: tcnksm/gox:1.5.1
+  volumes: ["/var/run/docker.sock", "/Users/faiq/go/src/github.com/wercker"]
```
Comment (termie) [inline wercker.yml]: the users faiq section probably needs to be passed in as an env var, which means these values will need to be interpolated, too

Comment (faiq) [inline wercker.yml]: it does- which means that core/config is going to need some new code.

Comment (termie) [inline wercker.yml]: also, this still is somewhat unlikely to work on the server where the working directories are randomized, so an env var will have to be passed in with the working dir  if we want to special case ourselves for testing

**Pull request closed without merging**